woocommerce/woocommerce-ios
# Introduce BetaFeature and automatic UI for beta features configuration

Closes: #7013


### Description

Following on the changes from #7012, this introduces a new `BetaFeature` where we can easily encapsulate all the details of any new experimental flag. It then replaces `BetaFeaturesViewController` with a new SwiftUI-based `BetaFeaturesConfigurationViewController`, which generates all its content automatically by iterating through all the defined beta features.

Note that the UI has changed slightly while porting this to SwiftUI, as having the descriptions on the footer felt more natural (and simpler) than adding a second row, and this is more of a proposal than a fully polished PR, but we can discuss and tweak that part.

I considered replacing the existing code that reads beta feature and deleting the relevant actions, but that would probably double the size the PR (at least), so maybe for a future one.



### Testing instructions


1. Go to Settings > Experimental features
2. Toggle the features and check that their values persist
3. Verify that toggling add-ons sends an analytics event

### Screenshots


Before|After
-|-
![Screen Shot 2022-05-31 at 17 22 22](https://user-images.githubusercontent.com/8739/171210017-9c802733-8888-4df9-8cad-62fc093feb2c.png)|![Screen Shot 2022-05-31 at 17 34 57](https://user-images.githubusercontent.com/8739/171213914-3dbc0179-c23b-4621-91ba-16c28953ee74.png)


---
